src/main/diagnostics/dockerCliSymlinks.ts

```diff
@@ -2,35 +2,100 @@ import fs from 'fs';
 import os from 'os';
 import path from 'path';
 
+import Electron from 'electron';
+
+import Logging from '@/utils/logging';
 import paths from '@/utils/paths';
 
 import type { DiagnosticsCategory, DiagnosticsChecker } from './diagnostics';
 
-const CheckDockerCLISymlinks: DiagnosticsChecker = {
-  id:         'RD_BIN_SYMLINKS',
-  category:   'Utilities' as DiagnosticsCategory,
-  applicable: ['darwin', 'linux'].includes(os.platform()),
+const console = Logging.diagnostics;
+
+export class CheckerDockerCLISymlink implements DiagnosticsChecker {
+  constructor(name: string) {
+    this.name = name;
+  }
+
+  readonly name: string;
+  get id() {
+    return `RD_BIN_DOCKER_CLI_SYMLINK_${ this.name.toUpperCase() }`;
+  }
+
+  readonly category = 'Utilities' as DiagnosticsCategory;
+  readonly applicable = ['darwin', 'linux'].includes(os.platform());
+  trigger?: ((checker: DiagnosticsChecker) => void) | undefined;
+
+  // For testing use
+  readonly readlink = fs.promises.readlink;
+  readonly access = fs.promises.access;
+
   async check() {
-    const allNames = await fs.promises.readdir(paths.integration, 'utf-8');
-    const names = allNames.filter(name => name.startsWith('docker-') && !name.startsWith('docker-credential-'));
+    const rdBin = path.join(paths.integration, this.name);
     const dockerCliPluginDir = path.join(os.homedir(), '.docker', 'cli-plugins');
-    const isInvalid = await Promise.all(names.map(async(name) => {
-      try {
-        const link = await fs.promises.readlink(path.join(dockerCliPluginDir, name));
-
-        return link !== path.join(paths.integration, name);
-      } catch (ex) {
-        // Is not a symlink, etc.
-        return true;
+    let passed = false;
+    let state;
+
+    function replaceHome(input: string) {
```

Comment:
Feel free to ignore this comment, but I'd prefer this to be a method of `CheckerDockerCLISymlink`, or maybe an external function. This would make the `check` function shorter, which would be helpful, I think.